tests/app.test.js

```diff
@@ -0,0 +1,97 @@
+const request = require('supertest');
+const app = require('../lib/app');
+
+jest.mock('../service/RickAndMortyApi.js', () => ({
+  getCharacter() {
+    return Promise.resolve({
+      name: 'Rick Sanchez',
+      species: 'Human',
+      status: 'Alive'
+    });
+  },
+  getCharacters() {
+    return Promise.resolve([
+      {
+        name: 'Rick Sanchez',
+        species: 'Human',
+        status: 'Alive'
+      },
+      {
+        name: 'Morty Smith',
+        species: 'Human',
+        status: 'Alive'
+      }
+    ]);
+  }
+}));
+
+
+describe('app', () => {
+
+  // it('can take a query string', () => {
+  //   return request(app)
+  //     .get('/you?name=unclebob')
+  //     .then(res => {
+  //       expect(res.body).toEqual({ test: 'hi there unclebob' });
+  //     });
+  // });
+
+  // it('gets a character by id', () => {
+  //   return request(app)
+  //     .get('/characters/1')
+  //     .then(res => {
+  //       expect(res.body).toEqual({
+  //         name: 'Rick Sanchez',
+  //         species: 'Human',
+  //         status: 'Alive'
+  //       });
+  //     });
+
+  // });
+
+  // it('can return all characters', () => {
+  //   return request(app)
+  //     .get('/characters')
+  //     .then(res => {
+  //       expect(res.text).toEqual(`<html>
+  //       <body>
+
+  //       </body>
+  //       </html>
+  //       `);
+  //     });
+  // });
+
+  // it('can post a note to a specific character', () => {
+  //   return request(app)
+  //     .post('/characters')
+  //     .send({ characterId: 1, note: 'Its pickle Rick' })
+  //     .then(res => {
+  //       expect(res.status).toEqual(204);
+  //     });
+  // });
+
+  // it('gets notes for a character', () => {
+  //   return request(app)
+  //     .post('/characters')
+  //     .send({ characterId: 1, note: 'Its pickle Rick' })
+  //     .then(() => {
+  //       return request(app)
+  //         .get('/characters/1');
+  //     }) 
+  //     .then(res => {
+  //       expect(res.text).toContain('Its pickle rick');
+  //     });
```

Comment:
```suggestion
  //     });
  });
```
Looks like you may have been tracking down a syntax error with all of this commented out code. If you add closing '})'s to this final test block the tests should run. As for the async issues, take a look at the [classwork](https://github.com/alchemy-fullstack-js-winter-2019/classwork/blob/master/week02_http-servers/class08_http-routing/test/app.test.js) to dissect the pieces you are missing.